docs/auth/example-lh-auth.test.js

```diff
@@ -0,0 +1,134 @@
+/** @typedef {import('./node_modules/lighthouse/types/lhr')} LH */
+
+const puppeteer = require('puppeteer');
+const lighthouse = require('lighthouse');
+const server = require('./server/server.js');
+
+const DEBUG_PORT = 8042;
+const SERVER_PORT = 8000;
+
+jest.setTimeout(30000);
+
+/**
+ * @param {string} url
+ * @return {Promise<LH.Result>}
+ */
+async function runLighthouse(url) {
+  const result = await lighthouse(url, {
+    port: DEBUG_PORT,
+    onlyCategories: ['seo'],
+  });
+  return result.lhr;
+}
+
+describe('my site', () => {
+  /** @type {import('puppeteer').Browser} */
+  let browser;
+  /** @type {import('puppeteer').Page} */
+  let page;
+
+  beforeAll(async () => {
+    server.listen(SERVER_PORT);
```

Comment:
probably want to await the callback for the server being up if this is code people will copy